### PR TITLE
fix: post signup

### PR DESCRIPTION
**Pull request by abdou059**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true
-  }
+  },
+  "cSpell.words": [
+    "asljfdkasj"
+  ]
```
Comment (LinaYahya) [inline .vscode/settings.json]:
```suggestion

```

File: server/src/validation/signupSchema.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  fullName: Yup
+    .string()
+    .required(),
+  password: Yup
+    .string()
+    .required('password is required')
+    .min(8, 'Password should be of minimum 8 characters length'),
+  phoneNumber: Yup
+    .number()
+    .required(),
```
Comment (LinaYahya) [inline server/src/validation/signupSchema.ts]: Where `fullName`, and `phoneNumber` validation messages?

File: server/src/services/checkEmail.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Customer } from '../db/models';
+
+const checkEmail = async ({ email }) => {
+  const gtiEmail = await Customer.findAll({
```
Comment (LinaYahya) [inline server/src/services/checkEmail.ts]:
```suggestion
  const getEmail = await Customer.findAll({
```

File: server/src/controllers/signup.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const {
+    fullName, email, password, phoneNumber,
+  } = req.body;
+  await signupSchema.validate({
+    fullName, email, password, phoneNumber,
+  });
+  const result = await checkEmail({ email });
+
+  if (result.length) {
+    throw createError(400, 'this email is signup');
```
Comment (LinaYahya) [inline server/src/controllers/signup.ts]:
```suggestion
    throw createError(400, 'this email is signup'); 
```
use a clear message

File: server/src/controllers/signup.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  const hashedPassword = await bcrypt.hash(password, 12);
+  const res = await signupUser({
```
Comment (LinaYahya) [inline server/src/controllers/signup.ts]:
```suggestion
  const user = await signupUser({
```

File: server/src/controllers/signup.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    email, password: hashedPassword, phoneNumber, fullName,
+  });
+  const token = await generateToken({ userId: res, role: 'user' });
```
Comment (LinaYahya) [inline server/src/controllers/signup.ts]: is `res` is really the useId, try to console `res`, to figure what is it?

File: server/src/controllers/signup.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  return { status: 200, msg: 'done!', data: token };
```
Comment (LinaYahya) [inline server/src/controllers/signup.ts]:
```suggestion
  return { status: 201, msg: 'done!', data: token };
```

Comment (LinaYahya) [reply server/src/controllers/signup.ts]: Also check the return structure How should it be

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: server/src/controllers/signup.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+  if (result.length) {
+    throw createError(400, 'this email is registered');
+  }
+  const hashedPassword = await bcrypt.hash(password, 12);
+  const user = await signupUser({
+    email, password: hashedPassword, phoneNumber, fullName,
+  });
+  const token = await generateToken({ userId: user, role: 'user' });
+  return { status: 201, msg: 'done!', data: token };
```
Comment (LinaYahya) [inline server/src/controllers/signup.ts]:
```suggestion
  return { status: 201, msg: 'done!', data: token };
```
data is not token, check `expressWrapper`

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: server/__tests__/cars.ts [hunk 9]
```diff
@@ -204,4 +204,20 @@ describe('/cars endpoint', () => {
...
+    expect(result.body.msg).toEqual('done!');
+  });
+  test('Signup', async () => {
+    const result = await request(app).post('/api/v1/auth/signup')
+      .send({
+        email: 'abdo123451@gmail.com', password: '123456789', phoneNumber: '0597111', fullName: 'abdo',
+      })
+      .expect(201);
+    expect(result.body.msg).toEqual('done!');
+  });
```
Comment (mai-mohammed) [inline server/__tests__/cars.ts:230]: if possible write tests for more conditions.

File: server/src/controllers/signup.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  const {
+    fullName, email, password, phoneNumber,
+  } = req.body;
+  await signupSchema.validate({
+    fullName, email, password, phoneNumber,
+  });
+  const result = await checkEmail({ email });
+
+  if (result.length) {
+    throw createError(400, 'this email is registered');
```
Comment (mai-mohammed) [inline server/src/controllers/signup.ts:19]: just return it and put the error in the returned object

**Review by mai-mohammed — CHANGES_REQUESTED** [review]

File: server/src/controllers/signup.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+const signupController = async (req:Request) => {
+  const {
+    fullName, email, password, phoneNumber,
+  } = req.body;
+  await signupSchema.validate({
+    fullName, email, password, phoneNumber,
+  });
+  const result = await checkEmail({ email });
+
+  if (result.length) {
```
Comment (husamkamal) [inline server/src/controllers/signup.ts:18]: add not 1

**Review by mai-mohammed — APPROVED** [review]

**Review by mai-mohammed — APPROVED** [review]

**Review by husamkamal — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by tariqabusada — APPROVED** [review]

**Pull request merged**